packages/functionals/botpress-qna/src/index.js

```diff
@@ -70,5 +74,67 @@ module.exports = {
         res.status(500).send(e.message || 'Error')
       }
     })
+
+    router.get('/csv', async (req, res) => {
+      const questions = await storage.getQuestions()
+      const rows = questions.flatMap(item => {
+        console.log(item)
+        const { questions, answer, action, redirectFlow, redirectNode } = item.data
+        return questions.map(question => ({
+          question,
+          action,
+          answer: action === 'text' ? answer : [redirectFlow, redirectNode].filter(Boolean).join('#')
+        }))
+      })
+      res.setHeader('Content-Type', 'text/csv')
+      res.setHeader('Content-disposition', `attachment; filename=qna_${moment().format('DD-MM-YYYY')}.csv`)
+      rows.forEach(({ question, action, answer }) => {
+        res.write(
+          [question, action, answer].map(field => '"' + field.toString().replace(/\"/g, '""') + '"').toString() + '\r\n'
+        )
+      })
+      res.end()
+    })
+
+    const upload = multer()
+    router.post('/csv', upload.single('csv'), async (req, res) => {
+      const parseFlow = str => {
+        const [redirectFlow, redirectNode] = str.split('#')
+        return { redirectFlow, redirectNode: redirectNode || '' }
+      }
+
+      if (yn(req.body.isReplace)) {
+        const questions = await storage.getQuestions()
+        await Promise.all(questions.map(({ id }) => storage.deleteQuestion(id)))
+      }
+
+      try {
+        const content = parse(req.file.buffer.toString(), { columns: ['question', 'action', 'answer'] })
+        await Promise.all(
```

Comment:
I'd rather not put the logic directly in the router, could you please extract the code to a module that does only one thing?

Reply:
OK, makes sense, moved parsing to separate module.